turbopack/crates/turbo-tasks-backend/src/database/db_invalidation.rs

```diff
@@ -0,0 +1,54 @@
+use std::{
+    fs::{self, read_dir},
+    io,
+    path::Path,
+};
+
+const INVALIDATION_MARKER: &str = "__turbo_tasks_invalidated_db";
+
+/// Atomically write an invalidation marker.
+///
+/// Because attempting to delete currently open database files could cause issues, actual deletion
+/// of files is deferred until the next start-up (in [`check_db_invalidation`]).
+///
+/// In the case that no database is currently open (e.g. via a separate CLI subcommand), you should
+/// call [`cleanup_db`] after this to eagerly remove the database files.
+///
+/// This should be run with the base (non-versioned) path, as that likely aligns closest with user
+/// expectations (e.g. if they're clearing the cache for disk space reasons).
+pub fn invalidate_db(base_path: &Path) -> io::Result<()> {
+    fs::write(base_path.join(INVALIDATION_MARKER), &[0u8; 0])?;
```

Comment:
should put something fun here.

"delort"
"persistence delenda est"
"last one out, please turn out the lights"
"This file intentionally left blank"

Reply:
I've been thinking about this, and I want to encode a "reason" in this file so that we can print a warning about persistent caching during the next start, which likely means this needs to be json.

We'll use this invalidation logic in panic handling, and in those cases, it might not be obvious to the user why their subsequent `next dev` start is so slow.

I'm not going to change this now, but I'll add a reason JSON field in a later PR, and when I do that, I can add an easter egg as a separate field.